.github/workflows/update_docs.yml

```diff
@@ -0,0 +1,61 @@
+name: Update Docs
+
+on:
+  pull_request_target:
+    branches:
+      - main
+  pull_request:
+
+env:
+  RAILS_ENV: test
+
+jobs:
+  labeler:
+    permissions:
+      contents: read
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Label PR
+        uses: actions/labeler@v4
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+  update-docs:
+    needs: labeler
+    if: contains(github.event.pull_request.labels.*.name, 'doc')
+    name: Run request specs and update docs
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    services:
+      db:
+        image: postgres:15
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_HOST_AUTH_METHOD: trust
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - name: Setup Database
+        run: |
+          bundle exec rake parallel:load_schema[8]
+      - name: Run tests and update docs
+        env:
+          PARALLEL_TESTS_CONCURRENCY: 8
```

Comment:
We use a step to detect how many cores the node has instead of hardcoding a value, in case it's helpful for you it's in the CI.yml.